### ISSUE TITLE
Task 2.1: Developing three personas

### ISSUE THREAD

**Issue by joshuag1214** [joshuag1214/ENPM613-Project#15]
Model your software’s users, by developing three personas: one for a typical user, one for
a special (non-typical) user, and another one for an attacker. The "special" user can be
e.g., a user with special needs or who uses a special technology (different than the typical
computer I/O devices, i.e., screen, keyboard, mouse, touchpad).
o Explain what needs/requirements for your system you have identified due to this
modeling.

**Comment by cpettit98** [comment]
<ins> **Persona 1 (Average User): Claire Johnson, Georgia Tech Civil Engineering Student Sophomore** </ins>
Claire is a 20 year old civil engineer student at Georgia Institute of Technology. She is a motivated, goal oriented student who enjoys the challenge of learning. Claire is also a very competitive person who partakes in trivia nights at the local pizza restaurant off campus. Unfortunately, Claires’ team has been getting destroyed in the sports portion of trivia nights so she has taken it upon herself to strengthen this area of their team. 
Claire loves trivia night with her friends, but often finds herself overwhelmed with school work and struggling to make the hang outs. She often takes her homework with her to the trivia nights or will skip to make sure she gets all of her work done. 
Claires’ goals are to: 
- Understand the rules, history, and current events of several olympic sports. 
- Learn quickly and on the go. Learn what she needs quickly and on the go. Ideally, she could go over sports facts while she is on the bus heading towards class. She needs to not spend too much time on this side interest so the website needs to be intuitive and easy to navigate. 


<ins>**Persona 2 (Blind User): Stephanie Hughes, School Teacher** </ins>
	Stephanie is an elementary school music teacher. She teaches a choir class and in her spare time she is an amateur author! Currently, she is writing a story about a young girl who is an aspiring olympic gymnast. The story follows the young girl's journey to the Olympics which is particularly unique, challenging, and inspiring because she is blind.
	Stephanie leads a relatively busy life because she is in charge of hundreds of children in the choir classes. Additionally, she leads certain extracurricular activities such as the school play. She tends to have busy seasons throughout the year when she doesn’t spend much time working on her book, and then works on her book mainly throughout the Summer. She has no issue taking her time doing in depth research on her book even if that research were to take years. She would much rather take 10 years writing an accurate book than the alternative. 
Stephanies’ goals are to: 
- Understand varying perspectives of gymnastics. Ideally she is looking for platforms that provide many different courses from different levels of qualified individuals: coaches, enthusiasts, athletes, etc. 
- Accuracy: Stephanie is looking to have a highly accurate source of information. She plans to cross reference all information that she includes in her book and is not keen on wasting her time on inaccurate information. 
- Usability: She is looking for a website that interfaces well with the VoiceOver function that she uses to navigate the internet and all digital resources.  



<ins>**Person 3 (Attacker): William Moore, 23 year old currently unemployed**</ins>
William Moore is a 23 year old who recently graduated with a degree in Computer Science. He has been struggling to find a job and has a lot of time on his hands. In his undergraduate studies he took several cybersecurity courses and still has some of the software on his device that can be used for brute force password attacks.
 He applied to a developer position for the SONIC web application and was not selected for the position. William decided that he was going to attempt to determine if any of the interviewers have accounts within the system that he could perform a brute force attack to determine their credentials. He will use BurpSuite to perform an intruder brute force attack. 
William’s goals are to: 
- Determine who has an account: If William can determine what email(s) are registered in the account he can choose which email to focus his resources on. 
- Determine credential information: Once he determines which accounts are registered he may be able to determine the password through a brute force attack. With this information he may change their password and account information to make their life more difficult. Additionally, he may be able to use the same credentials to access more important accounts of other websites such as banking information. 
- Take advantage of web application weaknesses: William is hoping that the website will reveal what emails are registered when you attempt to login. Additionally, if the password requirements are weak then he has a much better chance of determining a password. Lastly, he is looking to take advantage of a login feature that does not have a maximum password attempt lockout in place. He hopes that since the web application is relatively new the security features are not well developed and he will be able to get his revenge. 


@thefueley @joshuag1214 @trimpinm @adiwari Here are the user stories. I will add a blurb about features that we should implement from these users, but for now you all can provide comments and review them!

**Comment by thefueley** [comment]
I think the personas look great, @cpettit98! I only have a question about how we'll have to implement the site for persona 2. If we are to implement a voice over to read the site, do we also have to implement a voice input system? I am not sure how sites like that work, other than the haptic tap (Braille) input system. Maybe we can identify that input system as a future feature, if it's needed. Great work on the personas!

**Comment by CFPettit** [comment]
Thank you! So the VoiceOver is actually a common accessibility feature on laptops. I ended up searching how blind users navigate the internet and this was the first thing that came up. They can essentially tab through the page and their device will read out whatever they are currently on. To me this means that our website needs to be able to be tabbed through easily - so not too much embedded on a single page. Additionally, elements like buttons should have some type of label that can be read aloud. We would need to avoid unnamed or ambiguous buttons/features that only have visual queues (for example a question mark to represent FAQs). I assume that we will be able to name these features (probably through front end css element naming or something similar) to make this accesible if we did want a button that wasn’t labelled by being spelled out.
